### PR TITLE
Add gnocchi module

### PR DESCRIPTION
**Pull request by chenlizhong**
- GnocchiConf
- GnocchiMetricdLog

### CODE REVIEW

File: insights/parsers/tests/test_gnocchi.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    gnocchi_conf = GnocchiConf(context_wrap(GNOCCHI_CONF))
+    assert "storage" in gnocchi_conf.sections()
+    assert gnocchi_conf.get("indexer", "url") == "mysql+pymysql://gnocchi:exampleabckeystring@192.168.0.1/gnocchi?charset=utf8"
+    assert gnocchi_conf.getint("statsd", "flush_delay") == 10
+
+
+def test_metrics_log():
+    log = GnocchiMetricdLog(context_wrap(METRICD_LOG))
+    assert len(log.get('INFO')) == 3
+    assert len(list(log.get_after(datetime(2017, 04, 12, 19, 36, 38)))) == 1
```
Comment (bfahr) [inline insights/parsers/tests/test_gnocchi.py]: @chenlizhong would you mind to go ahead and add the tests for your examples?

File: insights/parsers/gnocchi.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        username=gnocchi
+        password=yourpassword23432
+        user_domain_name=Default
+        project_name=services
+        project_domain_name=Default
+
+
+    Examples:
+
+        >>> conf = shared[GnocchiConf]
```
Comment (xiangce) [inline insights/parsers/gnocchi.py]: `shared` is no longer used in 3.x. please modify the examples as per 3.0

File: insights/parsers/tests/test_gnocchi.py (hunk 1 above)
Comment (xiangce) [reply insights/parsers/tests/test_gnocchi.py]: @chenlizhong - I think @bfahr might mean you'd better add the test for the `Examples` in docstring to the unit test with e.g. `doctest`

File: insights/parsers/gnocchi.py (hunk 2 above)
Comment (psachin) [reply insights/parsers/gnocchi.py]: Hi @chenlizhong Sorry for jumping in but I see in some of parsers we have started including doctest. Please refer https://github.com/RedHatInsights/insights-core/pull/1092/files#diff-340ae940da9a54f85293da193efe7a17R54. This is a good time to include those in this parser as well.

**Pull request merged**